File: Front/src/types/application.ts
export interface ApplicationFormData {
  // Personal Information (Step 1)
  name: string;
  phone: string;
  email: string;
  city: string;
  state: string;
  zip: string;
  familyStatus: string;

  // Driving Experience (Step 2)
  drivingExperience: string;
  previousCompany: string;
  reasonForLeaving: string;
  felonies: string;
  drivingRecord: string;
  workedReefer: string;

  // Work Preferences (Step 3)
  dislike: string;
  hos: string;
  overnightPark: string;
  specialConsideration: string;
  onRoad: string;

  // Availability (Step 4)
  drugTest: string;
  securingUnloading: string;
  salaryExpectation: string;

  // Documents (Step 5)
  cdlFile: File | null;
  medicalCardFile: File | null;
}

export interface ApplicationSubmission extends Omit<ApplicationFormData, 'cdlFile' | 'medicalCardFile'> {
  // For database storage - files will be stored separately
  cdlFileUrl?: string;
  medicalCardFileUrl?: string;
  submittedAt: string;
  status: 'pending' | 'reviewed' | 'approved' | 'rejected';
  id: string;
}

File: Backend/HRDashboard.API/Models/Application.cs
namespace HRDashboard.API.Models
{
    public class Application
    {
        public string Id                  { get; set; } = "";
        public string CompanyId           { get; set; } = "";
        public DateTime SubmittedAt       { get; set; }
        public string Status              { get; set; } = "pending"; // pending | reviewed | approved | rejected

        // Personal Info
        public string Name                { get; set; } = "";
        public string Phone               { get; set; } = "";
        public string Email               { get; set; } = "";
        public string City                { get; set; } = "";
        public string State               { get; set; } = "";
        public string Zip                 { get; set; } = "";
        public string FamilyStatus        { get; set; } = "";

        // Driving Experience
        public string DrivingExperience   { get; set; } = "";
        public string PreviousCompany     { get; set; } = "";
        public string ReasonForLeaving    { get; set; } = "";
        public string Felonies            { get; set; } = "";
        public string DrivingRecord       { get; set; } = "";
        public string WorkedReefer        { get; set; } = "";

        // Work Preferences
        public string Dislike             { get; set; } = "";
        public string Hos                 { get; set; } = "";
        public string OvernightPark       { get; set; } = "";
        public string SpecialConsideration{ get; set; } = "";
        public string OnRoad              { get; set; } = "";

        // Availability
        public string DrugTest            { get; set; } = "";
        public string SecuringUnloading   { get; set; } = "";
        public string SalaryExpectation   { get; set; } = "";
    }
}
